frappe/utils/global_search.py

```diff
@@ -142,8 +142,8 @@ def _get_filters():
 				"name": frappe.db.escape(doc.name),
 				"content": frappe.db.escape(' ||| '.join(content or '')),
 				"published": published,
-				"title": frappe.db.escape(title or ''),
-				"route": frappe.db.escape(route or '')
+				"title": frappe.db.trim_varchar(frappe.db.escape(title or '')),
```

Comment:
for global search, the table creation is not based on db_schema, so you should create a `varchar_len` module level constant in this file and use ti.

Reply:
tests are failing too, no need to add a new method to `frappe.db`